src/la/arnoldi/arnoldi_data_manipulation.f90

```diff
@@ -0,0 +1,139 @@
+!--------------------------------------------------------------------------------------------------!
+! Copyright (C) by the DBCSR developers group - All rights reserved                                !
+! This file is part of the DBCSR library.                                                          !
+!                                                                                                  !
+! For information on the license, see the LICENSE file.                                            !
+! For further information please visit https://dbcsr.cp2k.org                                      !
+! SPDX-License-Identifier: GPL-2.0+                                                                !
+!--------------------------------------------------------------------------------------------------!
+
+#:include 'arnoldi.fypp'
+#:for nametype1, type_prec, real_zero, nametype_zero, type_nametype1, vartype in inst_params_1
+! **************************************************************************************************
+!> \brief ...
+!> \param arnoldi_data ...
+!> \param matrix ...
+!> \param max_iter ...
+! **************************************************************************************************
+  SUBROUTINE setup_arnoldi_data_${nametype1}$ (arnoldi_data, matrix, max_iter)
+     TYPE(arnoldi_data_type)                 :: arnoldi_data
```

Comment:
`intent` missing

Reply:
Why do we need intents if it works without them?